ipld/resolver/src/behaviour/discovery.rs

```diff
@@ -177,10 +184,20 @@ impl Behaviour {
     /// This seems to be the only way, because Kademlia rightfully treats
     /// incoming connections as ephemeral addresses, but doesn't have an
     /// alternative exchange mechanism.
-    pub fn add_identified(&mut self, peer_id: &PeerId, info: &Info) {
+    pub fn add_identified(&mut self, peer_id: &PeerId, info: Info) {
         if info.protocols.contains(&self.protocol_name) {
-            for addr in info.listen_addrs.iter().cloned() {
-                self.add_address(peer_id, addr);
+            // If we are still in the process of bootstrapping peers, buffer the incoming self-identify records,
```

Comment:
How this solve the problem? Even if we block during the bootstrap stage, an attacker could create a lot of sybils an try to connect to me to eventually evict the honest nodes from my routing tables. Or am I missing something?

Reply:
The paper linked to the issue is a good read, and it contains more countermeasures than this, such as not accepting too many IP addresses from the same subnet range. 

Actually, I have no idea if the current Kademlia protects against multiple peer IDs listening on the same address.

The issue here is how the k-table works. Say you put off bootstrapping for a while, and you get a huge influx of connections. The k-table has 256 buckets, one for each distance, with 20 places each; once those are filled, it will only put in more peers if someone from the 20 in the bucket becomes inresponsive. So, an attacker could send lots of connections it controls to take up as many of these buckets as possible, and by the time you have bootstrapped, the nodes you found would only be eligible if the attacker stops responding to pings. 

Because I think we work on `PeerId`, a hash of a public key, it's difficult to target a specific other peer to systematically fill up the k-table, but with a huge amount of random peer IDs, it should fill up eventually.

By not accepting incoming records until the bootstrap is over, we prioritise addresses we find on our own, rather than ones pushed upon us, so it should be a bit more trustworthy (as long as the bootstrap nodes are honest).